src/validation.cpp

```diff
@@ -2065,7 +2065,9 @@ bool ApplyTxInUndo(const CTxInUndo& undo, CCoinsViewCache& view, const COutPoint
         CCoinsModifier coins = view.ModifyCoins(out.hash);
         if (undo.nHeight != 0 ||
                 // Special-case genesis since nHeight is always 0, assume DB is clean
-                (Params().GenesisBlock().vtx[0]->GetHash() == out.hash && coins->IsPruned())) {
+                (Params().GenesisBlock().vtx[0]->GetHash() == out.hash && coins->IsPruned()) ||
+                // Special-case for initial issuance since being a single output nHeight is always 0
+                (Params().GenesisBlock().vtx.size() > 1 && Params().GenesisBlock().vtx[1]->GetHash() == out.hash && coins->IsPruned())) {
```

Comment:
I'm actually not totally sure what these lines are doing. Time to dig in.

Reply:
This didn't seem to be a bug before as the issuance had multiple outputs and the condition `undo.nHeight != 0` was always met. The added clause solves this, but it yea it would be a good idea to check this for any vtx in the genesis.

Reply:
are you interested in adding this check?

Reply:
I could do! Wasn't sure you were expecting it.

Reply:
Would prefer it; I feel if we merge as-is we'll completely forget about it and break something in the future.

Reply:
Sure I'll add it

Reply:
I've added the check! Had to rebase because same tests had changed, but it's fine now.